examples/siro_sandbox/sandbox_app.py

```diff
@@ -127,38 +128,54 @@ def set_debug_line_render(self, debug_line_render):
     def get_sim(self) -> Any:
         return self.env.task._sim
 
-    def draw_nav_hint_from_agent(self, end_pos, color):
-        agent_idx = 0
+    def draw_nav_hint_from_agent(self, end_pos, end_radius, color):
+        agent_idx = self.ctrl_helper.get_gui_controlled_agent_index()
+        assert agent_idx is not None
         art_obj = (
             self.get_sim().agents_mgr[agent_idx].articulated_agent.sim_obj
         )
         agent_pos = art_obj.transformation.translation
-        # todo: get forward_dir from FPS camera yaw, not art_obj.transformation
+        # get forward_dir from FPS camera yaw, not art_obj.transformation
         # (the problem with art_obj.transformation is that it includes a "wobble"
         # introduced by the walk animation)
-        forward_dir = art_obj.transformation.transform_vector(
-            mn.Vector3(0, 0, 1)
-        )
+        transformation = self.cam_transform or art_obj.transformation
+        forward_dir = transformation.transform_vector(-mn.Vector3(0, 0, 1))
         forward_dir[1] = 0
```

Comment:
We should probably normalize forward_dir after this. `forward_dir = forward_dir.normalized()`. The nav hint will display more consistently if the forward_dir is always the same length.

Side note: one should always beware when adding a call to `normalized` that the input vector won't ever be zero (otherwise you'll get a divide-by-zero). In this case, it's a somewhat safe assumption, since the camera is never supposed to be looking straight down from above.

Reply:
Added normalization step and tested locally, works as expected.